llvm/lib/Transforms/Vectorize/SandboxVectorizer/Legality.cpp

```diff
@@ -197,7 +198,15 @@ const LegalityResult &LegalityAnalysis::canVectorize(ArrayRef<Value *> Bndl) {
 
   // TODO: Check for existing vectors containing values in Bndl.
 
-  // TODO: Check with scheduler.
+  if (!SkipScheduling) {
+    // TODO: Try to remove the IBndl vector.
```

Comment:
What are the impediments for doing this? (other than changing the API of the scheduler, which should be fine to do in a follow up). If there are any long-term problems it would be nice to add some more detail in the TODO.

Reply:
I don' t think there are any issues, it should be just API changes.